### PR TITLE
LibVT: Miscellaneous features

### PR DESCRIPTION
**Pull request by BertalanD**
## LibVT: Add title stack support
This feature allows applications to reset the window title once they exited. It is used by `vim` if `set title` is enabled.

## LibVT: Implement Bracketed Paste Mode 
This mode allow us to escape any data that was not directly typed by the user. `vim` currently uses this. If we implement it in the shell, we could prevent newlines from being injected into the shell by pasting text or dragging files into it (see #7276).

### CODE REVIEW

File: Userland/Libraries/LibVT/Terminal.cpp [hunk 1]
```diff
@@ -936,10 +960,12 @@ void Terminal::execute_osc_sequence(OscParameters parameters, u8 last_byte)
     case 2:
         if (parameters.size() < 2)
             dbgln("Attempted to set window title without any parameters");
-        else
-            m_client.set_window_title(stringview_ify(1));
-        // FIXME: the split breaks titles containing semicolons.
-        // Should we expose the raw OSC string from the parser? Or join by semicolon?
+        else {
```
Comment (alimpfard) [inline Userland/Libraries/LibVT/Terminal.cpp:967]: Please wrap the `if` branch in braces if the `else` branch is in braces.

File: Userland/Libraries/LibVT/TerminalWidget.cpp [hunk 2]
```diff
@@ -1091,20 +1088,21 @@ void TerminalWidget::drop_event(GUI::DropEvent& event)
...
         for (auto& url : event.mime_data().urls()) {
             if (!first) {
-                write(m_ptm_fd, " ", 1);
+                constexpr StringView space = " ";
+                send_non_user_input(space.bytes());
                 first = false;
             }
             if (url.protocol() == "file")
-                write(m_ptm_fd, url.path().characters(), url.path().length());
+                send_non_user_input(url.path().to_byte_buffer());
```
Comment (alimpfard) [inline Userland/Libraries/LibVT/TerminalWidget.cpp:1103]: Could we use `url.path().bytes()` to avoid extra allocations?

Comment (alimpfard) [reply Userland/Libraries/LibVT/TerminalWidget.cpp:1103]: (Same below)

File: Userland/Libraries/LibVT/TerminalWidget.cpp [hunk 3]
```diff
@@ -1155,4 +1153,34 @@ void TerminalWidget::set_font_and_resize_to_fit(const Gfx::Font& font)
...
+        // We do not call write() separately for the control sequences and the data,
+        // because that would present a race condition where another process could inject data
+        // to prematurely terminate the escape. Could probably be solved by file locking.
+        Vector<u8> output;
+        output.ensure_capacity(leading_control_sequence.bytes().size() + bytes.size() + trailing_control_sequence.bytes().size());
+
+        // HACK: We don't have a `Vector<T>::unchecked_append(Span<T> const&)` yet :^(
+        [[maybe_unused]] auto rc1 = output.try_append(leading_control_sequence.bytes().data(), leading_control_sequence.bytes().size());
+        [[maybe_unused]] auto rc2 = output.try_append(bytes.data(), bytes.size());
+        [[maybe_unused]] auto rc3 = output.try_append(trailing_control_sequence.bytes().data(), trailing_control_sequence.bytes().size());
```
Comment (alimpfard) [inline Userland/Libraries/LibVT/TerminalWidget.cpp:1175]: There's no real reason to use `try_append()` here, there's an `append(const T*, size_t)` that does the same thing and asserts on failure.
your comments will still be valid.

Sidenote, can we not just write these separately?

Comment (BertalanD) [reply Userland/Libraries/LibVT/TerminalWidget.cpp:1175]: I'm not sure about the separate writing part. That could be a data race where some other process could maliciously inject the end marker.

I'll fix the `tri_append()` stuff.

Comment (alimpfard) [reply Userland/Libraries/LibVT/TerminalWidget.cpp:1175]: That's fair, I guess.

File: Userland/Libraries/LibVT/TerminalWidget.cpp (hunk 2 above)
Comment (BertalanD) [reply Userland/Libraries/LibVT/TerminalWidget.cpp:1103]: We can't skip the allocation in `URL.to_string()`, right?

Comment (alimpfard) [reply Userland/Libraries/LibVT/TerminalWidget.cpp:1103]: Nope, `url.to_string()` will end up allocating a string regardless, but it's not nice to allocate more and then copy into it :P

**Pull request merged**